launch_ros/launch_ros/default_launch_description.py

```diff
@@ -94,12 +96,12 @@ def _run(self):
 
     def _function(self, context: launch.LaunchContext):
         try:
-            rclpy.init(args=context.argv)
+            rclpy.init(args=context.argv, context=self.__rclpy_context)
```

Comment:
This is only needed if the context is not already initialized. I'd recommend requiring that the context that was passed in is initialized, and also require that shutting it down is the responsibility of the caller.